### PR TITLE
Auto receiver connect

### PR DESCRIPTION
**Pull request by mikkelspring**

### CODE REVIEW

File: src/main/scala/ylabs/messaging/Client.scala [hunk 1]
```diff
@@ -22,6 +22,9 @@ object Client {
 
   case class User(value: String) extends AnyVal
   case class Password(value: String) extends AnyVal
+  case class Domain(value: String) extends  AnyVal
```
Comment (mpollmeier) [inline src/main/scala/ylabs/messaging/Client.scala:25]: hmm, looks like you went a bit overboard with all these case classes. How about we only keep `case class User(name: String, domain: Option[String])`?

File: src/main/scala/ylabs/messaging/Client.scala [hunk 2]
```diff
@@ -97,18 +108,26 @@ class Client extends FSM[State, Context] {
     case Event(Messages.RegisterEventListener(actor), ctx) ⇒
       stay using ctx.copy(eventListeners = ctx.eventListeners + actor)
 
-    case Event(msg: Messages.ListenerEvent, ctx) ⇒
+    case Event(msg: Messages.ListenerEvent, ctx @ Context(Some(connection), chats, _)) ⇒
```
Comment (mpollmeier) [inline src/main/scala/ylabs/messaging/Client.scala]: if you replace the _ in the pattern match with `listeners` then you don't need `ctx @`

**Comment by mikkelspring** [comment]
Why do you feel it's better with fewer case classes? I feel like it's harder to use now because not only does it change how user is created (no longer can just do @ domain) but it's now less clear which functions need the user without the domain and which need it with. 

Give it a look and see.

**Comment by mpollmeier** [comment]
Ok, after reading the whole (old) changeset I see what you're saying and agree. 
LGTM - do you just wanna revert ce55150 and merge?

**Pull request merged**